src/ug_sensors/ug_sensors/depth_sensor.py

```diff
@@ -0,0 +1,144 @@
+"""
+
+
+def main(args=None):
+    rclpy.init(args=args)
+    node = depthSensorNode()
+    rclpy.spin(node)
+    rclpy.shutdown()
+
+
+"""
+    Node that periodically publishes depth sensor data to the /sensors/depth topic
+
+    TODO: Error handling
+"""
+
+
+class depthSensorNode(Node):
+    def __init__(self):
+        super().__init__("depth_sensor")
+
+        self.depth_sensor = depthSensor(fresh_water=False)
+
+        # Create publisher for data
+        self.depth_data_pub_ = self.create_publisher(
+            depth_sensor_interface, "/sensors/depth", 10
+        )
+
+        # Every half second publish data to the topic
+        # TODO: How often does this need to be updated?
+        self.timer_ = self.create_timer(0.5, self.publish_depth_data)
+
+        # Advertise that the node has started
+        # TODO: Standardize log messages with other team members
+        self.get_logger().info("Depth sensor node started")
+
+    # Publish the data
+    def publish_depth_data(self):
+        msg = depth_sensor_interface()
+
+        msg.data = self.depth_sensor.read_depth()
+
+        # Publish data
+        self.depth_data_pub_.publish(msg)
+
+        # Log data (Debugging)
+        # self.get_logger().info("Hello " + str(self.data_))
+
+
+"""
+    Simple class to manage the sensor driver and provide some abstraction
+
+    TODO: Error handling. What happens if __sensor.read() is false?
+
+    __sensor.read() must be called whenever new data is requested,
+    but if both depth and temp are wanted simultaniously, theres no
+    reason to call read() twice.
+"""
+
+
+class depthSensor:
+    # Initalize the physical sensor, returns whether init was successful
+    def __init__(self, fresh_water=True) -> bool:
+        self.__sensor = (
+            sensor_driver.MS5837_30BA()
+        )  # Default I2C bus is 1 (Raspberry Pi 3)
+
+        # We must initialize the sensor before reading it
+        if not self.__sensor.init():
+            # Sensor could not be initialized
+            return False
+
+        # We have to read values from sensor to update pressure and temperature
+        if not self.__sensor.read():
+            # Sensor read failed!
+            return False
+
+        # Freshwater is default, but the glider would usually be in saltwater
+        if fresh_water:
+            self.__sensor.setFluidDensity(sensor_driver.DENSITY_FRESHWATER)
+        else:
+            self.__sensor.setFluidDensity(sensor_driver.DENSITY_SALTWATER)
+
+        return True
+
+    # Only call read() once for all data
+    def read_depth_temp(self) -> tuple[float, float]:
+        if self.__sensor.read():
+            return (self.__sensor.depth(), self.__sensor.temp())
+
+    # Read the depth in mBar from the sensor
+    def read_depth(self) -> float:
+        if self.__sensor.read():
+            return self.__sensor.depth()
+
+    # Read the temperature in Celcius from the sensor
+    def read_temp(self) -> float:
```

Comment:
Make sure you are able to handle errors effectively whenever reading from a sensor and log it accordingly with good desccription